crates/red_knot_python_semantic/src/types/class.rs

```diff
@@ -460,6 +496,28 @@ impl<'db> ClassLiteral<'db> {
         })
     }
 
+    #[salsa::tracked(cycle_fn=legacy_generic_context_cycle_recover, cycle_initial=legacy_generic_context_cycle_initial)]
```

Comment:
Is it worth to have this as a dedicated query, considering that `explicit_bases` is a query (which caches gives us caching for both `legacy_generic` and `inherited_legacy`

Reply:
Hmm, several commits back on this branch I had to add this because of the cycle handling. But removing it no longer causes any cycle-related panics...

We're also post-processing the results of `explicit_bases` in both cases, and I thought the extra cache would prevent us from having to duplicate that work. Are you thinking that the overhead of the extra Salsa query will dominate that post-processing work?

Reply:
> Are you thinking that the overhead of the extra Salsa query will dominate that post-processing work?

Hard to tell but quiet possibly. It's also a balance between memory usage and perf where fewer queries means less memory usage.


I did note that you call a method that uses the AST node in at least one of the two cases. So removing the tracked might, after all, not be possible.